### ISSUE TITLE
Add `Maximum size (MiB)` filter

### ISSUE THREAD

**Issue by Korb** [peps1/airdcpp-auto-downloader#29]
For music and movies, size is often proportional to quality. This is not always the case, but in many cases it is. Without the ability to evaluate a file in terms of bitrate, duration, resolution, and codec, ADC users are often forced to choose something larger, hoping that what they find will not disappoint.

However, when searching for documents (e-books, magazines, comics, other journalism), this principle often fails, and instead of tens and hundreds of megabytes of text documents, airdcpp-auto-downloader downloads gigabytes of movies. And the "File type" filter does not help.

In theory, I could list a hundred extensions-exceptions. If only extension supported it. Or indicate dozens of formats for saving text information and archives. But it would be much easier to use a size limit (depending on the expected size of the files being searched for).

**Comment by peps1** [comment]
Makes sense, don't remember why I didn't add the max size filter.

Not sure I understand what you're saying about extensions, they should support adding many extensions using separated by `;`
Though, looking at the API docs right now `extensions` can't be used together with the `file_type` filter though, I'll need to make that clear in the UI.

**Comment by Korb** [comment]
> Not sure I understand what you're saying about extensions, they should support adding many extensions using separated by `;`

I meant extensions that should not be downloaded as part of the current task. For example, if I'm looking for a specific book, I would add the following file extensions to the exclusions:

`3gp;ape;asf;avi;divx;dv;fla;flv;idx;ifo;m1v;m2ts;m2v;m4v;mkv;mov;mp4;mpeg…`

However, the module settings do not allow you to specify **exclusion** extensions, only **inclusion** extensions:

> File extensions (optional)
> Separate extensions with ; (example: exe;iso;img)

**Comment by peps1** [comment]
Correct, the API on airdc++ side only supports extensions to include, see api doc here: https://airdcpp.docs.apiary.io/#reference/searching/search-instances/create-instance

How many extensions would you have to include for your book filter? I’d think that would be only a couple 🤔

**Comment by Korb** [comment]
> How many extensions would you have to include for your book filter? I’d think that would be only a couple 🤔

azw;azw3;cb7;cba;cbr;cbt;cbz;ceb;djvu;doc;docx;epub;fb2;htm;html;ibook;ibooks;kf8;kfx;koi;lit;lrf;lrx;mobi;opf;oxps;pdb;pdf;pdg;plz;prc;rb;rtf;snb;tcr;text;tr2;tr3;txt;txtz;utf;xeb;xps + all archive extensions [1] + multi-volume archive extensions. I don't want to miss anything, but I don't want to do too much manual processing either.

[1] 7z;ace;afa;alz;apk;ar;arc;arj;ark;b1;b6z;ba;bh;bz2;cab;car;cd;cdx;cfs;cpio;crx;dar;dd;dgc;dump;ear;ecc;ecsbx;esd;ff;gca;gz;ha;hki;ice;ima;img;iso;jar;kgb;lbr;lha;lpaq;lz;lz4;lzh;lzma;lzo;lzx;makeself;mar;mou;mpkg;pak;paq;paq6;paq7;paq8;par;par2;partimg;pax;pea;phar;pim;pit;pmp;q;qda;rar;rev;rk;rz;s7z;sbx;sea;sen;sfark;shar;shk;sit;sitx;sqx;swm;sz;tar;tbz2;tgz;tlz;txz;uc;uc0;uc2;uca;ucn;ue2;uha;ur2;war;wim;xar;xf;xp3;xz;yz1;z;zip;zipx;zoo;zpaq;zst;zz

**Comment by peps1** [comment]
Right, that is a lot. Thanks for these details.
